dashboard/test/controllers/omniauth_callbacks_controller_test.rb

```diff
@@ -477,18 +493,123 @@ def setup_should_connect_provider(user, timestamp = Time.now)
 
       user.reload
       assert_response :success
-      assert_equal 1, user.authentication_options.count
-      new_auth_option = user.authentication_options.last
-      assert_equal user, new_auth_option.user
-      assert_equal User.hash_email('new@email.com'), new_auth_option.hashed_email
-      assert_equal 'google_oauth2', new_auth_option.credential_type
-      assert_equal user.uid, new_auth_option.authentication_id
-      expected_auth_data = {
-        oauth_token: auth.credentials.token,
-        oauth_token_expiration: auth.credentials.expires_at,
-        oauth_refresh_token: auth.credentials.refresh_token
+      assert_oauth_auth_option(user, auth)
+    end
+  end
+
+  test 'connect_provider: creates new windowslive auth option for signed in user' do
+    user = create :user, :multi_auth_migrated, uid: 'some-uid'
+    auth = OmniAuth::AuthHash.new(
```

Comment:
I'd suggest extracting helpers for the fake AuthHashes (e.g. `fake_google_auth_hash`, `fake_windowslive_auth_hash`) for readability, and you may subsequently be able to DRY up the implementations of those methods.  It seems especially appropriate since we don't care about the particulars of this hash until the `assert_oauth_auth_option` method anyway.